tests/Feature/Controllers/ControllerTestCase.php

```diff
@@ -2,11 +2,29 @@
 
 namespace Tests\Feature\Controllers;
 
+use Illuminate\Database\Eloquent\Factories\Factory;
 use Tests\TestCase;
 
 use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Support\Str;
 
 abstract class ControllerTestCase extends TestCase
 {
     use RefreshDatabase;
+
+    public function factory($repo = null): Factory
+    {
+        $modelName = $repo ? $repo->class() : $this->guessModel();
+
+        return Factory::factoryForModel($modelName);
+    }
+
+    protected function guessModel()
+    {
+        $className = Str::of(class_basename($this))
+            ->before('ControllerTest')
+            ->prepend('\\App\\');
+
+        return (string) $className;
+    }
```

Comment:
@cbj4074 is there another way we can get the model? (maybe adapt the way used in RepositoryTestCase where each instance has to set repo?)

Reply:
I don't think there's a better way to get the model than what you've already done here, coupled with forcing each implementing test to define that `setRepository()` method.